### ISSUE TITLE
Fails to compile in Android due to name conflict with softfloat3e

### ISSUE THREAD

**Issue by Biswa96** [bochs-emu/Bochs#300]
* configure command:

```
../configure \
--build=aarch64-unknown-linux-android \
--prefix=/data/data/com.termux/files/usr \
--with-x11 \
--enable-usb \
--enable-ne2000 \
--enable-clgd54xx \
--enable-voodoo \
--enable-sb16 \
--enable-es1370 \
--with-sdl2 \
--enable-usb-ohci \
--enable-usb-xhci \
--enable-usb-ehci \
--enable-all-optimizations
```

* compiler version:

```
$ clang --version
clang version 18.1.2
Target: aarch64-unknown-linux-android24
Thread model: posix
InstalledDir: /data/data/com.termux/files/usr/bin
```

* compiler error:

```
g++ -c -I. -I../. -Iinstrument/stubs -I../instrument/stubs -D_FILE_OFFSET_BITS=64 -D_LARGE_FILES -g -O2 -D_FILE_OFFSET_BITS=64 -D_LARGE_FILES    -I/data/data/com.termux/files/usr/include/SDL2 -I/data/data/com.termux/files/usr/include -DNO_SHARED_MEMORY -D_REENTRANT -D_THREAD_SAFE    -DBX_SHARE_PATH='"/data/data/com.termux/files/usr/share/bochs"' ../main.cc -o main.o
In file included from ../main.cc:45:
In file included from /data/data/com.termux/files/usr/include/SDL2/SDL.h:38:
In file included from /data/data/com.termux/files/usr/include/SDL2/SDL_cpuinfo.h:81:
In file included from /data/data/com.termux/files/usr/lib/clang/18/include/arm_neon.h:38:
/data/data/com.termux/files/usr/lib/clang/18/include/arm_vector_types.h:17:15: error: typedef redefinition with different types ('float' vs 'struct f32_t')
   17 | typedef float float32_t;
      |               ^
../cpu/fpu/../softfloat3e/include/softfloat_types.h:60:3: note: previous definition is here
   60 | } float32_t;
      |   ^
```

**Comment by stlintel** [comment]
Eventually when the transition will be complete I would keep old names for float type
I.e. float32_t will be back float32
until then the situation won't be resolved

**Comment by stlintel** [comment]
Fixed in a05f9c85d3444d7a7501a6505a037b1ff6adc5b1
